### ISSUE TITLE
Custom selector strategies cannot be used when chaining queries with the new Async API

### ISSUE THREAD

**Issue by drawkh** [webdriverio/webdriverio#7258]
**Environment:**
 - **WebdriverIO version:** 7.9.1
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** async
 - **Node.js version:** 14.17.3
 - **NPM version:** 6.14.13
 - **Browser name and version:** Chrome 92
 - **Platform name and version:** Linux

**Describe the bug**
Custom selector strategies cannot be used when chaining queries with the new Async API. The tests fail indicating that the strategy was not found.

**To Reproduce**

1. Add a custom selector strategy:
```ts
browser.addLocatorStrategy('testId', (selector, root) => (root ?? document).querySelectorAll(`[data-testid="${selector}"]`));
```
2. Have an HTML page containing those elements in the body:
```html
<div>
  <button data-testid="my-testid">Test</button>
</div>
```
3. Use the custom selector strategy by chaining queries:
```ts
await browser.$('div').custom$('testId', 'my-testid');
```

**Expected behavior**
The element with the attribute `data-testid="my-testid"` should be found.

**Log**
```
[chrome 92.0.4515.131 linux #0-0] No strategy found for testId
[chrome 92.0.4515.131 linux #0-0] Error: No strategy found for testId
[chrome 92.0.4515.131 linux #0-0]     at Element.custom$ (/home/ggautier/Projects/client/node_modules/webdriverio/build/commands/browser/custom$.js:31:15)
[chrome 92.0.4515.131 linux #0-0]     at Element.wrapCommandFn (/home/ggautier/Projects/client/node_modules/@wdio/utils/build/shim.js:103:38)
[chrome 92.0.4515.131 linux #0-0]     at processTicksAndRejections (internal/process/task_queues.js:95:5)
[chrome 92.0.4515.131 linux #0-0]     at async Element.wrapCommandFn (/home/ggautier/Projects/client/node_modules/@wdio/utils/build/shim.js:103:29)
[chrome 92.0.4515.131 linux #0-0]     at async SearchCreationPage.open (/home/ggautier/Projects/client/tests/e2e/pageobjects/search/creation.page.ts:40:9)
[chrome 92.0.4515.131 linux #0-0]     at async Context.<anonymous> (/home/ggautier/Projects/client/tests/e2e/specs/search/creation.spec.ts:27:9)
[chrome 92.0.4515.131 linux #0-0]     at async Context.executeAsync (/home/ggautier/Projects/client/node_modules/@wdio/utils/build/shim.js:284:16)
[chrome 92.0.4515.131 linux #0-0]     at async Context.testFrameworkFnWrapper (/home/ggautier/Projects/client/node_modules/@wdio/utils/build/test-framework/testFnWrapper.js:52:18)
```

**Comment by christian-bromann** [comment]
If someone is interested to look into this, I assume we propagate the wrong context when executing the command in our [`shim`](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-utils/src/shim.ts#L154).

**Comment by christian-bromann** [comment]
Fixed in #7397
